test/unit/puppetlabs/services/jruby/jruby_testutils.clj

```diff
@@ -29,6 +28,26 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;; JRubyPuppet Test util functions
 
+(schema/defn ^:always-validate
+create-mock-pool-instance :- JRubyInstance
```

Comment:
To expand on this a bit, I think it was the timed deref in `class-info-updated-after-cache-flush-during-prior-request` that was failing (https://github.com/jpinsonault/puppetserver/blob/3d23836dcdf8126300b2f2e9016e4a21c646d052/test/integration/puppetlabs/services/master/environment_classes_int_test.clj#L691) - if we bumped it from 1000 ms to 2000 ms it more reliably passed. We hyphothesized that the reason it was taking longer/timing out was because if we just mocked out the `get-initialize-pool-instance-fn` then the instance was still being created, just not loading puppet code, but the instance being created was enough to cause the deref to time out.